target/xilinx/constraints/carfield_islands.tcl

```diff
@@ -0,0 +1,108 @@
+# Copyright 2024 ETH Zurich and University of Bologna.
+# Solderpad Hardware License, Version 0.51, see LICENSE for details.
+# SPDX-License-Identifier: SHL-0.51
+#
+# Cyril Koenig <cykoenig@iis.ee.ethz.ch>
+
+####################
+# Clock generators #
+####################
+
+set SOC_TCK 20
+
+## Generate a clock based on the default clk_sel value for a clock mux
+## @param clk_sel_path path to the clk_sel register
+## @param clk_sel_val default clk_sel value
+proc handle_domain_clock_mux { clk_sel_path clk_sel_val clk_name } {
+  upvar SOC_TCK SOC_TCK
```

Comment:
Is this to link the `SOC_TCK` with the one defined in the `xdc`? Because in both the `xdc` and `tcl` we initialize `SOC_TCK`, so why are we also aliasing them?

I think the idea is that we are defining it once and aliasing it, right?